### DISCUSSION TITLE
Using tRPC for server to server communication

### DISCUSSION THREAD

**Discussion by danielyogel** [trpc/trpc#1566] [Q&A] [answered]
Is there something that prevents using a tRPC client inside a node server, to allow it communicate with another node server?

**Comment by KATT** [comment, marked as answer]
Not afaik and I think some people do this!

**Reply by soulsam480** [reply]
Is there a way a to call a tRPC mutation inside server ?

**Reply by ecwyne** [reply]
@soulsam480 you can call `router.createCaller(...).mutation(...)` and pass in whatever context is required.

**Reply by soulsam480** [reply]
Nice.

**Reply by Tobjoern** [reply]
But how does this work exactly?
Since the actual url isn't used, I assume the mutation is simply run on the server, that is making the call.

But what if the mutation route requires environment variables etc.?

**Reply by hxgdzyuyi** [reply]
I suggest using JSON-RPC.
